consensus/istanbul/core/core.go

```diff
@@ -343,14 +331,14 @@ func (c *core) startNewRound(round *big.Int) {
 	// New snapshot for new round
 	c.updateRoundState(newView, c.valSet, roundChange)
 	// Calculate new proposer
-	c.valSet.CalcProposer(lastProposer, newView.Round.Uint64())
+	c.valSet.CalcProposer(lastAuthor, newView.Round.Uint64())
 	c.setState(StateAcceptRequest)
-	if roundChange && c.isProposer() && c.current != nil && request != nil {
+	if roundChange && c.current != nil && c.valSet.IsProposer(c.address) && request != nil {
```

Comment:
Is it guaranteed that `c.valset` is not nil?  The original `isProposer` method checked for that.

Reply:
valSet is set at the beginning so it won't be nil.

Either way I'm planning to remove this all together in a future commit. This was prework, to expose that it actually comes from the valset, not core